dali/operators/reader/loader/loader.h

```diff
@@ -469,17 +573,25 @@ class Loader {
   // The epoch number the next returned sample belongs to,
   // tracked only if checkpointing is enabled
   int consumer_epoch_;
+  // Batch size
+  int max_batch_size_;
   // Number of data shards that were actually read by the reader
+  // TODO(skarpinski) Make it private to prevent ReadSample from depending on it
   int virtual_shard_id_;
   // Keeps pointer to the last returned sample just in case it needs to be cloned
-  LoadTargetSharedPtr last_sample_ptr_tmp;
+  IndexedLoadTargetSharedPtr last_sample_ptr_tmp;
 
   struct ShardBoundaries {
     Index start;
     Index end;
   };
 
   std::deque<ShardBoundaries> shards_;
+
+ private:
+  bool initial_buffer_filled_ = false;
+  // Counts how many samples the reader have read already from this epoch
+  Index read_sample_counter_;
```

Comment:
```suggestion
  Index read_sample_counter_ = 0;
```
...and get rid of the initializer in the constructor?